calico_node/filesystem/startup.py

```diff
@@ -259,9 +260,10 @@ def main():
     # Write a startup environment file containing the IP address that may have
     # just been detected.
     # This is required because the confd templates expect to be able to fill in
-    #  the IP by fetching it from the environment.
+    # some templates by fetching them from the environment.
     with open('startup.env', 'w') as f:
-        f.write("IP=" + ip)
+        f.write("export IP=%s\n" % ip)
```

Comment:
Yeah, I'm looking into it.  I'm not sure what's going on.

`calicoctl node --ip=<something>` will set the `IP` env var when starting the container (via a `-e` flag)

The case I'm talking about is when no `-e IP` flag is passed to Docker (i.e `calico/node` must auto-detect).

It looks like when I start the container with `-e IP` ( butno address specified), the auto-detection works just fine.  When I start it with no `-e` flag for the `IP`var, the auto-detection works but the value is not propagated to confd, which fails when trying to fill in the bird template here https://github.com/projectcalico/calico-containers/blob/master/calico_node/filesystem/etc/calico/confd/templates/bird.cfg.mesh.template#L5